Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -866,12 +865,7 @@ static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<Character
         }
         // Consume the ']'
         buffer.advance();
-        // Check for unknown annotations with critical flag
-        // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
-        // step 4(a)(ii)(2)(d)(i)
-        if (isCritical)
-            return std::nullopt;
-        return CalendarRecord { /* isCritical = */ false, /* isUnknown = */ true, { } };
+        return CalendarRecord { isCritical, /* isUnknown = */ true, { } };
```

Comment:
Can we get an enum for isUnknown? In general, I like seeing named enums instead of bools, although I don't think this should block this patch.

Reply:
Sure, I can add that in the next patch. Would you want `isCritical` to be an enum as well?

Reply:
→ https://github.com/WebKit/WebKit/pull/44132